File: semana-1/short-to-url.rb
prefix = ENV.fetch('APP_WEBSITE','shorter.app')

SHORT_URL_KEY_POSITION = 0
ORIGINAL_URL_POSITION = 1

puts "Digite a url curta que deseja localizar:"
# Ref https://www.rubyguides.com/2019/10/ruby-chomp-gets/
shortUrl = gets.chomp



File.foreach("urls.csv") do |line|    
    #Ref https://ruby-doc.com/core/String.html#method-i-delete_prefix
    keyFromUser = shortUrl.delete_prefix("#{prefix}/")
    # Ref https://ruby-doc.com/core/String.html#method-i-split
    breakedLine = line.split(',')
    keyFromFile = breakedLine[SHORT_URL_KEY_POSITION]

    if ( keyFromUser == keyFromFile)
        puts "A sua url original é: #{breakedLine[ORIGINAL_URL_POSITION]}" 
        # Ref https://www.rubyguides.com/2019/03/ruby-exit/
        exit 0
    end
end if File.file?("urls.csv")

puts "Essa url não tem correspondência no nosso banco de dados"
exit 1

File: semana-1/short-to-url.js
const fs = require('fs');

const prefix = process.env.APP_WEBSITE || 'shorter.app'
const SHORT_URL_KEY_POSITION = 0
const ORIGINAL_URL_POSITION = 1


// Ref https://nodejs.org/api/fs.html#fsreadfilesyncpath-options
// Ref https://geshan.com.np/blog/2021/10/nodejs-read-file-line-by-line/#read-file-sync
// Ref https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/String/split
const readData = () => fs.readFileSync('urls.csv', 'utf-8')
.split(/\r?\n/)

// Ref https://nodejs.dev/learn/accept-input-from-the-command-line-in-nodejs
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout,
  });

  
readline.question('Digite a url curta que deseja localizar:', shortUrl => {
    readline.close();    
    const lines = readData()
    
    //Ref https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/String/replace
    const keyFromUser = shortUrl.replace(`${prefix}/`, '')

    // Ref https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach
    const lineShort = lines
    .map((line) => line.split(','))
    .find(breakedLine => keyFromUser == breakedLine[SHORT_URL_KEY_POSITION] )


    if (lineShort) {
        console.log(`A sua url original é: ${lineShort[ORIGINAL_URL_POSITION]}`)
        process.exit(0)
    }
  
    console.log("Essa url não tem correspondência no nosso banco de dados")
    process.exit(1)
});